src/Feldspar/Compiler/Imperative/Frontend.hs

```diff
@@ -508,8 +508,9 @@ decodeType = goL []
              Just (n, t') = decodeLen t
              (ts, t'') = structGo n t' []
              structGo 0 s acc = (reverse acc, s)
-             structGo n' ('_':s) acc = structGo (n' - 1) s' (ts':acc)
-                      where (ts', s') = go s
+             structGo n' ('_':s) acc = structGo (n' - 1) s' (replicate m ts' ++ acc)
```

Comment:
The comment "Encoded format is:" should be updated for this change.

Reply:
Yes indeed.